src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -300,9 +307,14 @@ public static Imports FromGlobalUsings(CSharpCompilation compilation)
 
         private void MarkImportDirective(CSharpSyntaxNode directive, bool callerIsSemanticModel)
         {
-            if (directive != null && _compilation != null && !callerIsSemanticModel)
+            MarkImportDirective(_compilation, directive, callerIsSemanticModel);
+        }
+
+        private static void MarkImportDirective(CSharpCompilation compilation, CSharpSyntaxNode directive, bool callerIsSemanticModel)
+        {
+            if (directive != null && compilation != null && !callerIsSemanticModel)
```

Comment:
I've converted the condition into an assert, but I'll merge it as part of a subsequent PR.